Formula/wordle.rb

```diff
@@ -0,0 +1,17 @@
+class Wordle < Formula
+  desc "Play wordle in command-line"
+  homepage "https://git.hanabi.in/wordle-cli"
+  url "https://git.hanabi.in/repos/wordle-cli.git", revision: "06f22bd2c41032901c514dfb74509392456fefe3", tag: "v1.0.0"
+  license "AGPL-3.0-only"
+  head "git.hanabi.in/repos/wordle-cli.git", tag: "v1.0.0", revision: "06f22bd2c41032901c514dfb74509392456fefe3"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w"), "src/main.go"
+  end
+
+  test do
+    system "test", "-f", "#{bin}/wordle"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@SMillerDev -- this formulæ does not take in an input file, it does not create an output file, it is an interactive game, which reads from standard input and writes to standard output.  I think doing something like `binary --about` is discouraged, but then, it is just an interactive cli game.  Will `binary --about` or `binary --help` be sufficient for a test in this case?

Reply:
You can probably use Ruby's [PTY module](https://ruby-doc.org/stdlib-2.6.3/libdoc/pty/rdoc/PTY.html) to interact with this software. `grep` for `require "pty"` in this repo for many examples.